bgpd/bgp_nht.c

```diff
@@ -581,6 +581,34 @@ void bgp_delete_connected_nexthop(afi_t afi, struct peer *peer)
 	}
 }
 
+static void bgp_bnc_mark_nht_important(struct bgp_nexthop_cache *bnc, struct zapi_route *nhr)
+{
+	struct bgp *bgp = bnc->bgp;
+	afi_t afi = family2afi(nhr->prefix.family);
+	struct bgp_table *table = bgp->rib[afi][nhr->safi];
+	struct bgp_dest *dest;
+
+	if (nhr->type != ZEBRA_ROUTE_BGP)
+		return;
+
+	/*
+	 * If the old resolved prefix is the same as the new resolved prefix
+	 * nothing to mark here, move along
+	 */
+	if (prefix_same(&bnc->resolved_prefix, &nhr->prefix))
+		return;
+
+	dest = bgp_afi_node_get(table, afi, nhr->safi, &bnc->resolved_prefix, NULL);
```

Comment:
Can `table` be NULL somehow here?

Reply:
I doubt it, but I'll check and return otherwise for safety